File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        // from root -> get root.left and root.right
        // add them to a list, and then append that list to main list
        // from that root.left -> get root.left and root.right and same for root.right

        List<List<Integer>> result = new ArrayList<>();

        if (root == null) {
            return result;
        }

        Queue<TreeNode> queue = new LinkedList<>();
        queue.add(root);

        while (!queue.isEmpty()){
            int levelSize = queue.size();

            List<Integer> subLis = new ArrayList<>();

            for (int i =0; i < levelSize; i++){
                TreeNode curr = queue.poll();
                subLis.add(curr.val);
                    
                if(curr.left != null) {
                    queue.add(curr.left);
                }
                
                if(curr.right != null) {
                    queue.add(curr.right);
                }
            }
            result.add(subLis);


        }
    return result;
    }

}

File: Data Structures & Algorithms/lowest-common-ancestor-of-a-binary-tree-iii/submission-0.py
"""
# Definition for a Node.
class Node:
    def __init__(self, val):
        self.val = val
        self.left = None
        self.right = None
        self.parent = None
"""

class Solution:
    def lowestCommonAncestor(self, p: 'Node', q: 'Node') -> 'Node':
        ancestors = set()

        node = p
        while node:
            ancestors.add(node)
            node = node.parent

        while q:
            if q in ancestors:
                return q 
            q = q.parent